src/vm/arm64/pinvokestubs.S

```diff
@@ -32,6 +32,15 @@
         // if null goto stub generation
         cbz                 x9, LOCAL_LABEL(\__PInvokeStubFuncName\()_0)
 
+        .if (\CalliHelper == 1)
+                //
+                // We need to distinguish between a MethodDesc* and an unmanaged target for the CalliHelper.
+                // The way we do this is to shift the managed target to the left by one bit and then set the
+                // least significant bit to 1.  This works because MethodDesc* are always 8-byte aligned.
+                //
+                lsl         \HiddenArg, \HiddenArg, #1
+                orr         \HiddenArg, \HiddenArg, #1
+        .endif
```

Comment:
This piece is needed for Windows as well.  It also looks like it is needed for arm32.  @adityamandaleeka has offered to take care of Windows for me.

Reply:
Since it is difficult for me to compile for Windows, I much appreciate the offer to do this.  The *.asm syntax is different and it would take lots of tries.  Thanks @adityamandaleeka

Reply:
I took a look at the code in dllimport where the corresponding right-shifting is being done. It's under !ARM and !x86. Maybe we can just make it explicitly for AMD64 only and avoid the shifting back and forth on ARM64 too?

Reply:
I can't find the right shift code in dllimport.

I do see this.
```
    // The MethodDesc pointer may in fact be the unmanaged target, see PInvokeStubs.asm.
    if (pMD == NULL || (UINT_PTR)pMD & 0x1)
```

Reply:
This is what I was referring to: https://github.com/dotnet/coreclr/blob/master/src/vm/dllimport.cpp#L2272

Reply:
thanks